python/deltalake/schema.py

```diff
@@ -17,34 +17,91 @@
 DataType = Union["PrimitiveType", "MapType", "StructType", "ArrayType"]
 
 
-def delta_arrow_schema_from_pandas(
-    data: "pd.DataFrame",
-) -> Tuple[pa.Table, pa.Schema]:
-    """
-    Infers the schema for the delta table from the Pandas DataFrame.
-    Necessary because of issues such as:  https://github.com/delta-io/delta-rs/issues/686
-
-    Args:
-        data: Data to write.
+### Inspired from Pola-rs repo - licensed with MIT License, see licenses folder.###
+def _convert_pa_schema_to_delta(
```

Comment:
Perhaps we should add additional unit tests to enforce that these conversions are expected behavior now?

Reply:
Done, I've added many test cases for:

- large -> normal
- normal -> large
- mixed large/normal -> large
- mixed large/normal -> normal
- uint -> int
- nested lists with string/binary/int
- timestamp -> timestamp(us)
- struct with nested fields